Renaming Does Not Work on Windows

According to [this](https://stackoverflow.com/questions/64800455/how-to-change-app-icon-and-app-name-for-flutter-desktop-application), not `windows/CMakeLists.txt` but `windows/runner/main.cpp` has to be changed in order to rename the app on Windows.

Also, changing the BINARY_NAME in `windows/CmakeLists.txt` causes the following build issue to occur:

```
Building Windows application...
CMake Error at runner/CMakeLists.txt:9 (add_executable):
  The target name "X X X" is reserved or not valid for certain
  CMake features, such as generator expressions, and may result in undefined
  behavior.


CMake Error at CMakeLists.txt:41 (target_compile_features):
  Cannot specify compile features for target "X X X" which is not
  built by this project.
Call Stack (most recent call first):
  runner/CMakeLists.txt:21 (apply_standard_settings)

.
.
.

Error: Unable to generate build files
```

> According to [this](https://stackoverflow.com/questions/64800455/how-to-change-app-icon-and-app-name-for-flutter-desktop-application), not `windows/CMakeLists.txt` but `windows/runner/main.cpp` has to be changed in order to rename the app on Windows.
> 
> Also, changing the BINARY_NAME in `windows/CmakeLists.txt` causes the following build issue to occur:
> 
> ```
> Building Windows application...
> CMake Error at runner/CMakeLists.txt:9 (add_executable):
>   The target name "X X X" is reserved or not valid for certain
>   CMake features, such as generator expressions, and may result in undefined
>   behavior.
> 
> 
> CMake Error at CMakeLists.txt:41 (target_compile_features):
>   Cannot specify compile features for target "X X X" which is not
>   built by this project.
> Call Stack (most recent call first):
>   runner/CMakeLists.txt:21 (apply_standard_settings)
> 
> .
> .
> .
> 
> Error: Unable to generate build files
> ```

Ok, I will look into this issue ASAP.

--- a/lib/constants.dart
+++ b/lib/constants.dart
@@ -13,7 +13,7 @@ const iosDebugPlistPath = "ios/Runner/Info-Debug.plist";
 const iosReleasePlistPath = "ios/Runner/Info-Release.plist";
 /// File path of web/maniafest.json for web.
 const webManifestPath = "web/manifest.json";
-/// File path of windows/CMakeLists.txt for windows.
-const windowsCmakeListsPath = "windows/CMakeLists.txt";
+/// File path of windows/runner/main.cpp for windows.
+const windowsMainCppPath = "windows/runner/main.cpp";
 /// File path of linux/my_application.cc for linux.
 const linuxApplicationPath = "linux/my_application.cc";

--- a/lib/rename_app.dart
+++ b/lib/rename_app.dart
@@ -50,7 +50,7 @@ class RenameApp {
 
   static void mac(String mac) {}
   static Future<void> windows(String windows) async {
-    if (await Utils.fileNotExists(windowsCmakeListsPath)) {
+    if (await Utils.fileNotExists(windowsMainCppPath)) {
       Utils.printNoConfigFound('windows');
       return;
     }
@@ -59,7 +59,7 @@ class RenameApp {
       return;
     }
 
-    await Utils.renameWindows(windowsCmakeListsPath, windows);
+    await Utils.renameWindows(windowsMainCppPath, windows);
   }
 
   static Future<void> linux(String linux) async {

--- a/lib/utils.dart
+++ b/lib/utils.dart
@@ -82,9 +82,9 @@ class Utils {
 
   static Future<void> renameWindows(String filePath, String appName) async {
     String data = await File(filePath).readAsString();
-    var appNameLine = RegExp(r'set\(BINARY_NAME.*').firstMatch(data)?.group(0);
+    var appNameLine = RegExp(r'if \(!window\.Create\(L".*?"').firstMatch(data)?.group(0);
     if (appNameLine != null) {
-      data = data.replaceAll(appNameLine, """set(BINARY_NAME "$appName")""");
+      data = data.replaceAll(appNameLine, """if (!window.Create(L"$appName\"""");
     }
     await saveFile(filePath, data);
     printFinishMessage('windows');

